solutions/usaco-787.mdx

```diff
@@ -87,4 +86,85 @@ int main() {
 ```
 
 </CPPSection>
+<JavaSection>
+```java
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.StringTokenizer;
+
+public class rental {
+    static class Shop {
+        int price;
+        int numGal;
+        public Shop (int numGal, int price) {
+            this.price = price;
+            this.numGal = numGal;
+        }
+    }
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new FileReader("rental.in"));
+        PrintWriter out = new PrintWriter("rental.out");
+        StringTokenizer s = new StringTokenizer(br.readLine());
+        int n = Integer.parseInt(s.nextToken());
+        int m = Integer.parseInt(s.nextToken());
+        int r = Integer.parseInt(s.nextToken());
+        Integer [] production = new Integer [n];
+        for (int i = 0; i < n; i++) {
+            production[i] = Integer.parseInt(br.readLine());
+        }
+        Shop [] shops = new Shop [m];
+        for (int i = 0; i < m; i++) {
+            s = new StringTokenizer(br.readLine());
+            shops[i] = new Shop (Integer.parseInt(s.nextToken()),Integer.parseInt(s.nextToken()));
+        }
+        Integer [] rent = new Integer [r];
```

Comment:
1. why Integer instead of int
2. `Integer[] rent = new Integer[r];` - no whitespace for the brackets

Reply:
I use Integer instead of int because I use Collections.reverseOrder() in Arrays.sort().

Reply:
can you add a comment explaining that